File: apps/sivic_mobile/lib/compartido/modelos/usuario.modelo.dart
class Usuario {
  final int    usuarioId;
  final String nombre;
  final String email;
  final String rol;

  const Usuario({
    required this.usuarioId,
    required this.nombre,
    required this.email,
    required this.rol,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
    usuarioId: json['usuario_id'] as int,
    nombre:    json['nombre'] as String,
    email:     json['email'] as String,
    rol:       json['rol'] as String,
  );

  bool get esAdmin => rol == 'admin';
}

File: apps/sivic_web/src/app/compartido/modelos/usuario.modelo.ts
export type RolUsuario = 'admin' | 'guardia';

export interface Usuario {
  usuario_id: number;
  nombre: string;
  email: string;
  rol: RolUsuario;
  created_at?: string;
}

export interface SesionUsuario {
  usuario: Usuario;
  access: string;
}

export interface CredencialesLogin {
  email: string;
  password: string;
}
